### DISCUSSION TITLE
Re-encode images in place

### DISCUSSION THREAD

**Discussion by maufl** [immich-app/immich#14780] [Q&A] [answered]
Hey, thanks for this great software! I was wondering whether it is possible to re-encode images in a library in place, or whether that would be equivalent to re-importing them.
I'm thinking of switching from JPEG to JXL but I've already indexed a lot of images and would like to avoid having to re-upload them.

**Comment by bo0tzz** [comment, marked as answer]
You shouldn't mess with the files on disk, that'll confuse Immich. You'll need to remove and reupload them, or you could write a script involving the https://immich.app/docs/api/replace-asset endpoint.

**Reply by maufl** [reply]
Thanks, that's good to know :+1:
